pkg/recipes/configloader/environment.go

```diff
@@ -101,14 +101,15 @@ func getConfiguration(environment *v20231001preview.EnvironmentResource, applica
 		return nil, ErrUnsupportedComputeKind
 	}
 
-	providers := environment.Properties.Providers
-	if providers != nil {
-		if providers.Aws != nil {
-			config.Providers.AWS.Scope = to.String(providers.Aws.Scope)
-		}
-		if providers.Azure != nil {
-			config.Providers.Azure.Scope = to.String(providers.Azure.Scope)
-		}
+	env, err := environment.ConvertTo()
+	if err != nil {
+		return nil, err
+	}
+	if environment.Properties.Providers != nil {
+		config.Providers = env.(*datamodel.Environment).Properties.Providers
+	}
+	if environment.Properties.RecipeConfig != nil {
+		config.RecipeConfig = env.(*datamodel.Environment).Properties.RecipeConfig
```

Comment:
I see that we do this twice: `env.(*datamodel.Environment)`. Should we do it once before the if cases and use that within them?

Reply:
done